diff/walking/differ.go

```diff
@@ -56,6 +57,38 @@ func NewWalkingDiff(store content.Store) diff.Comparer {
 	}
 }
 
+func ReadonlyMounts(mounts []mount.Mount) []mount.Mount {
```

Comment:
Probably could make sense to add a GoDoc string to this function to describe what it does.


Slightly wondering;

- this function is passed to `mount.WithTempMount()`
- both https://github.com/moby/moby/pull/44964 and https://github.com/moby/buildkit/pull/1100 will use this function separate from `diff/walking` (?)

Would it make sense to;

- Move this function to the `mount` package
- Perhaps even (haven't looked closely) to have a `mount.WithReadOnlyTempMount()` (variant of `mount.WithTempMount()` that performs this out of the box
- ^^ Which could be "instead" of exporting `ReadOnlyMounts()` if that would work

./cc @rumpl

Reply:
Yeah, I thought about moving it to `mount` while I was doing it (and meant to leave a comment here asking for ideas about it but forgot 😅 ). I like it, and the semantics of `mount.WithReadOnlyTempMount()` too.

Reply:
@thaJeztah yes indeed, this looks like it could go into the `mount` package.

Reply:
+1 to another package or not having this function exported

Reply:
I took all your input, did a little refactor and introduced `mount.WithReadonlyTempMount`. PTAL when you get a chance :)